contracts/SpokePool.sol

```diff
@@ -924,14 +924,9 @@ abstract contract SpokePool is
             revert NotExclusiveRelayer();
         }
 
-        // Don't let caller override destination chain ID so they can't attempt a replay attack of an identical
-        // fill that was destined for a different chain. This is required because the relayData
-        // is used to form the relayHash which this contract uses to uniquely ID relays.
-        if (relayData.destinationChainId != chainId()) revert InvalidChainId();
-
         USSRelayExecutionParams memory relayExecution = USSRelayExecutionParams({
             relay: relayData,
-            relayHash: keccak256(abi.encode(relayData)),
+            relayHash: keccak256(abi.encode(relayData, chainId())),
```

Comment:
We follow the same pattern for computing the relayHash in several places and rely on it always being formatted the same. but abi.encode() won't give us this guarantee (i.e. if we forget to supply `chainId()` in one place). wdyt about a helper function that requires that we pass the chainId?

(nb. this error is tbf something that _should_ be caught in test)

Reply:
I was trying to reduce gas costs but I'm not sure it does anything so I'll implement the internal hash helper!